packages/react-icons/src/components.test.tsx

```diff
@@ -0,0 +1,30 @@
+import * as icons from '@fluentui/react-icons';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ReactIs from 'react-is';
+
+const iconsDir = 'src/components';
+
+describe('Icon components', () => {
+  fs.readdirSync(iconsDir).forEach(file => {
+    const componentName = path.basename(file, '.tsx');
+    // tslint:disable-next-line:no-any
+    const IconComponent = (icons as any)[componentName];
+
+    test(`${componentName} has 'Icon' suffix in the name`, () => {
+      expect(componentName.endsWith('Icon')).toEqual(true);
+    });
+
+    test(`${file} is exported in package`, () => {
+      expect(IconComponent).toBeDefined();
+    });
+
+    test(`${file} is a valid component`, () => {
+      expect(ReactIs.isValidElementType(IconComponent)).toEqual(true);
+    });
+
+    test(`${file} matches displayName`, () => {
```

Comment:
May be good to add a few more conformance tests like it can take a `className` prop, what about passing along `data-` attributes, do we support `as` (I don't think so but can't remember.)

Reply:
Will add the `data-` attributes and `className` tests, and you are right we do not support `as` prop.